python/ray/data/block.py

```diff
@@ -153,41 +134,31 @@ def __call__(self, __arg: T) -> Union[U, Iterator[U]]:
 
 
 def _apply_strict_mode_batch_format(given_batch_format: Optional[str]) -> str:
-    ctx = ray.data.DataContext.get_current()
-    if ctx.strict_mode:
-        if given_batch_format == "default":
-            given_batch_format = "numpy"
-        if given_batch_format not in VALID_BATCH_FORMATS_STRICT_MODE:
-            raise StrictModeError(
-                f"The given batch format {given_batch_format} is not allowed "
-                f"in Ray 2.5 (must be one of {VALID_BATCH_FORMATS_STRICT_MODE})."
-            )
+    if given_batch_format == "default":
+        given_batch_format = "numpy"
+    if given_batch_format not in VALID_BATCH_FORMATS_STRICT_MODE:
+        raise StrictModeError(
```

Comment:
I'm wondering if we should change these all to DeprecationWarning in 2.6, since it's a bit odd to have a "mode" that cannot be disabled.

Reply:
Seems like most of the errors are raised due to bad inputs, so I replaced `StrictModeError` to `ValueError`